File: 백준/Bronze/5585. 거스름돈/거스름돈.py
def count_coins(paid_amount):
    change = 1000 - paid_amount
    coins = [500, 100, 50, 10, 5, 1]
    coin_count = 0
    
    for coin in coins:
        coin_count += change // coin
        change %= coin
    
    return coin_count

# 입력
paid_amount = int(input())

# 잔돈의 개수 출력
print(count_coins(paid_amount))

File: 백준/Bronze/5585. 거스름돈/거스름돈.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // 입력
        int paidAmount = scanner.nextInt();
        
        // 잔돈의 개수 출력
        System.out.println(countCoins(paidAmount));
        
        scanner.close();
    }
    
    public static int countCoins(int paidAmount) {
        int change = 1000 - paidAmount;
        int[] coins = {500, 100, 50, 10, 5, 1};
        int coinCount = 0;
        
        for (int coin : coins) {
            coinCount += change / coin;
            change %= coin;
        }
        
        return coinCount;
    }
}
